### PR TITLE
[DEVX-1555] Add concurrency shard option

### PR DESCRIPTION
**Pull request by tianfeng92**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexplischke** [comment]
It'd be good to add an explanation of what it means to split by concurrency.

**Comment by tianfeng92** [comment]
@alexplischke Can you help to take look at this?

File: docs/web-apps/automated-testing/cypress/yaml.md [hunk 1]
```diff
@@ -725,8 +725,11 @@ Controls whether or not tests are run in headless mode.
 ### `shard`
 <p><small>| OPTIONAL | STRING |</small></p>
 
-When sharding is configured, saucectl automatically splits the tests (e.g., by spec) so that they can easily run in parallel.
-Selectable values: `spec` to shard by spec file. Remove this field or leave it empty `""` for no sharding.
+
+When sharding is configured, saucectl automatically splits the tests (e.g., by spec or concurrency) so that they can easily run in parallel.
+For sharding by concurrency, saucectl splits test files into serveral groups by concurrency. Then it starts job for each group.
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/cypress/yaml.md:730]:
```suggestion
For sharding by concurrency, saucectl splits test files into several groups (the number of groups is determined by the concurrency setting). Each group will then run as an individual job.
```

Comment (alexplischke) [reply docs/web-apps/automated-testing/cypress/yaml.md:730]: Copy paste the above suggestion to playwright and testcafe as well 🙏

**Pull request merged**